language/stdlib/modules/transaction_fee_collection.mvir

```diff
@@ -0,0 +1,58 @@
+module TransactionFeeCollection {
+  import 0x0.LibraCoin;
+  import 0x0.Block;
+
+  // A pot of libra that is collected per transaction, and that will then be distributed to
+  // validators. This also tracks the current height of the blockchain in order to prevent double
+  // distributions per-block.
+  resource CollectedFees {
+    fees: LibraCoin.T,
+    current_height: u64,
+  }
+
+  public initialize() {
+    assert(get_txn_sender() == 0xA550C18, 0);
+    move_to_sender<CollectedFees>(CollectedFees {
+        fees: LibraCoin.zero(),
+        current_height: Block.get_current_height(),
+      });
+    return;
+  }
+
+  public pay_transaction_fee(fee_to_deposit: LibraCoin.T) acquires CollectedFees {
+    let collected_fees: &mut Self.CollectedFees;
+    collected_fees = borrow_global_mut<CollectedFees>(0xA550C18);
+    LibraCoin.deposit(&mut move(collected_fees).fees, move(fee_to_deposit));
+    return;
+  }
+
+  public extract_transaction_fees(): LibraCoin.T * u64 acquires CollectedFees {
```

Comment:
Is returning a tuple here necessary? It seems like it's always the case that `ret.1 == LibraCoin.value(&ret.0)`, so you could get away with just returning the `LibraCoin`.